stl/inc/ranges

```diff
@@ -45,10 +47,12 @@ namespace ranges {
 
         // clang-format off
         template <_Not_same_as<ref_view> _OtherRng>
-            requires convertible_to<_OtherRng, _Rng&> && requires { _Rvalue_poison(_STD declval<_OtherRng>()); }
-        constexpr ref_view(_OtherRng&& _Other) noexcept(noexcept(static_cast<_Rng&>(_STD declval<_OtherRng>())))
-            // clang-format on
-            : _Range{_STD addressof(static_cast<_Rng&>(_STD forward<_OtherRng>(_Other)))} {}
+        constexpr ref_view(_OtherRng&& _Other) noexcept(
+            noexcept(static_cast<_Rng&>(_STD forward<_OtherRng>(_Other)))) // strengthened
+            requires convertible_to<_OtherRng, _Rng&> && requires {
+                _Rvalue_poison(static_cast<_OtherRng&&>(_Other));

```

Comment:
This is intentional: I try to avoid instantiating and/or specializing templates during overload resolution for throughput. In this case, that means `static_cast<T&&>(t)` in the constraints instead of `std::forward<T>(t)` that is used in `noexcept`-specifiers and function template bodies which are instantiated after overload resolution. There's an argument to be made that the difference is anomalous enough to offset any readability benefit.